lib/api/controllers/securityController.js

```diff
@@ -597,19 +614,161 @@ class SecurityController {
+      .then(() => createMethod(request, request.input.body, request.input.resource._id, request.input.args.strategy));
+  }
+
+  /**
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  updateCredentials(request) {
+    const
+      updateMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'update'),
+      validateMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'validate');
+
+    assertHasStrategy(request);
+    assertIsStrategyRegistered(this.kuzzle, request);
+    assertHasId(request);
+    assertHasBody(request);
+
+
+    return validateMethod(request, request.input.body, request.input.resource._id, request.input.args.strategy, true)
+      .then(() => updateMethod(request, request.input.body, request.input.resource._id, request.input.args.strategy));
+  }
+
+  /**
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  hasCredentials(request) {
+    const existsMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'exists');
+
+    assertHasStrategy(request);
+    assertIsStrategyRegistered(this.kuzzle, request);
+    assertHasId(request);
+
+
+    return existsMethod(request, request.input.resource._id, request.input.args.strategy);
+  }
+
+  /**
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  validateCredentials(request) {
+    const validateMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'validate');
+
+    assertHasStrategy(request);
+    assertIsStrategyRegistered(this.kuzzle, request);
+    assertHasBody(request);
+
+    // _id can be null on purpose
+    return validateMethod(request, request.input.body, request.input.resource._id, request.input.args.strategy, false);
+  }
+
+  /**
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  deleteCredentials(request) {
+    const deleteMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'delete');
+
+    assertHasStrategy(request);
+    assertIsStrategyRegistered(this.kuzzle, request);
+    assertHasId(request);
+
+
+    return deleteMethod(request, request.input.resource._id, request.input.args.strategy)
+      .then(() => ({acknowledged: true}));
+  }
+
+  /**
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  getCredentials(request) {
+    assertHasStrategy(request);
+    assertIsStrategyRegistered(this.kuzzle, request);
+    assertHasId(request);
+
+    if (this.kuzzle.pluginsManager.hasStrategyMethod(request.input.args.strategy, 'getInfo')) {
+      const getInfoMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'getInfo');
+
+      return getInfoMethod(request, request.input.resource._id, request.input.args.strategy);
+    }
+
+    return Bluebird.resolve({});
+  }
+
+  /**
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  getCredentialsById(request) {
```

Comment:
is the `id` parameter given here the user's identifier for the strategy (examples : the login for local strategy) ? or the kuzzle user's uid ?